Sources/gg.datagram.web-requests.sdPlugin/app.js

```diff
@@ -12,14 +12,45 @@ function connected(jsn) {
  *     settings: {
  *       url?: string,
  *       method?: string,
+ *       contentType?: string|null,
+ *       headers?: string|null,
  *       body?: string|null,
  *     }
  *   },
  * }} data
  */
 function sendHttp(data) {
-    const { url, method, body } = data.payload.settings;
-    log('sendHttp', { url, method, body });
+    const { url, method, contentType, headers, body } = data.payload.settings;
+    log('sendHttp', { url, method, contentType, headers, body });
+
+    let defaultHeaders = contentType ? {
+        'Content-Type':  contentType
+    } : {};
+    let inputHeaders = {};
+
+    if (headers) {
+        const headersArray = headers.split(/\n/);
+
+        for (let i = 0; i < headersArray.length; i += 1) {
+            if (headersArray[i].includes(':')) {
+                const [headerItem, headerItemValue] = headersArray[i].split(/:(.+)/);
+                const trimmedHeaderItem = headerItem?.trim();
+                const trimmedHeaderItemValue = headerItemValue?.trim();
+
+                if (trimmedHeaderItem && trimmedHeaderItemValue) {
```

Comment:
Empty header values are technically valid, so I'd like to allow them.

Keeping the code you had before and changing the split regex to `/:(.*)/` would be a better fix.

Reply:
Fetch doesn't seem to like empty header values though? Allowed empty header values with:

<img width="602" alt="image" src="https://user-images.githubusercontent.com/4265429/152709072-03bd00f4-5192-4b39-b351-b417f64a87c6.png">

but fetch doesn't like it

<img width="569" alt="image" src="https://user-images.githubusercontent.com/4265429/152709086-7dd6ad7b-ab05-4d60-93a2-61f73aa377e0.png">

Reply:
Ah. there's a bug with the initial regex anyway, the `:` is in the `headerItem`